projects/lexical_shared/lib/lexical/document/path.ex

```diff
@@ -52,26 +54,37 @@ defmodule Lexical.Document.Path do
     convert_separators_to_native(path)
   end
 
+  def from_uri(%URI{scheme: "untitled", path: name}) do
+    "untitled:" <> name
```

Comment:
It should be noted that 'untitled' is not an official specification of lsp. There is a related discussion here of neovim, but I'm not sure if there's a better solution.

* https://github.com/neovim/neovim/issues/21276

Reply:
No exact scheme (`http`, `sftp`, even `file`, etc) of URI is officially supported by LSP, the spec officially supports any valid URI as defined by rfc3986.

Turns out `untitled:Name` [is actually a valid URI](https://datatracker.ietf.org/doc/html/rfc3986#section-3), *which I didn't know until a few hours ago!* :D

Passing it to `URI.parse/1` even works:
```elixir
iex(1)> URI.parse("untitled:Untitled-1")
%URI{
  scheme: "untitled",
  userinfo: nil,
  host: nil,
  port: nil,
  path: "Untitled-1",
  query: nil,
  fragment: nil
}
```

For now I think matching against `untitled:` is clean enough, and it shores up a major usability/adoption issue.